x-pack/plugins/infra/server/routes/snapshot/lib/apply_metadata_to_last_path.ts

```diff
@@ -59,6 +59,9 @@ export const applyMetadataToLastPath = (
       if (inventoryFields.os) {
         lastPath.os = get(firstMetaDoc, inventoryFields.os) as string;
       }
+      if (inventoryFields.cloudProvider) {
+        lastPath.cloudProvider = get(firstMetaDoc, inventoryFields.cloudProvider) as string;
```

Comment:
Perhaps here it could already be the `CloudProvider` I mentioned above. Wdyt?

Reply:
Good point! The only downside of this strict typing is that this should be updated every time we have a new cloud provider added and can cause some errors if we miss it. But I don't think we will need to update it many times in reality.